updateinfo_monitor/schemas.py

```diff
@@ -0,0 +1,210 @@
+import logging
+from datetime import datetime
+from pathlib import Path
+
+import createrepo_c
+import gi
+from pydantic import AnyHttpUrl, BaseModel, Field
+
+gi.require_version("Modulemd", "2.0")
+from gi.repository import Modulemd
+
+
+class Repository(BaseModel):
+    id: int | None = None
+    name: str
+    arch: str = ""
+    description: str = ""
+    url: AnyHttpUrl
+    exclude_arch: list[str] = Field(default_factory=list)
+    debuginfo: bool = False
+    repomd_etag: str | None = None
+    repomd_checksum: str | None = None
+    check_ts: datetime | None = None
+    last_error: str | None = None
+    check_result: dict | None = Field(default_factory=dict)
+
+    class Config:
+        orm_mode = True
+
+    @property
+    def full_name(self) -> str:
+        return f"{self.name}.{self.arch}"
+
+    def dict_for_create(self):
+        return {
+            "name": self.name,
+            "arch": self.arch,
+            "url": self.url,
+            "debuginfo": self.debuginfo,
+        }
+
+
+class Distribution(BaseModel):
+    name: str
+    version: str
+    arches: list[str]
+    repositories: list[Repository]
+    sources: list[Repository]
+
+
+class RepomdRecord(BaseModel):
+    data_type: str
+    checksum: str
+    checksum_type: str
+    checksum_open: str | None = None
+    checksum_open_type: str | None = None
+    timestamp: int
+    location_href: str
+    size: int
+    size_open: int
+    path: Path | None = None
+
+
+class Package(BaseModel):
+    name: str
+    epoch: str
+    version: str
+    release: str
+    arch: str
+    location_href: str = ""
+    cr_nevra: str = ""
+
+    @staticmethod
+    def from_cr_package(cr_package: createrepo_c.Package) -> "Package":
+        return Package(
+            name=cr_package.name,
+            epoch=cr_package.epoch,
+            version=cr_package.version,
+            release=cr_package.release,
+            arch=cr_package.arch,
+            location_href=cr_package.location_href,
+            cr_nevra=cr_package.nevra(),
+        )
+
+    @staticmethod
+    def from_cr_updatepackage(
+        cr_package: createrepo_c.UpdateCollectionPackage,
+    ) -> "Package":
+        return Package(
+            name=cr_package.name,
+            epoch=cr_package.epoch,
+            version=cr_package.version,
+            release=cr_package.release,
+            arch=cr_package.arch,
+        )
+
+    @property
+    def nevra(self) -> str:
+        nevra = self.cr_nevra
```

Comment:
`epoch` can be empty, like `None`. Please either convert it to 0 explicitly or do not use it if it's empty.

Reply:
`createrepo_c` do not return `None` if `epoch` is empty, I've double check this